### ISSUE TITLE
Text rendering: line spacing problem

### ISSUE THREAD

**Issue by gabm** [femtovg/femtovg#197]
I have a problem where vertical line spacing is not sufficient. I followed the examples on text renderings.. here is my setup

My canvas has a scale that is applied "top-level". Its `0.35` in this instance.

```
        let mut transform = Transform2D::identity();
        transform.scale(self.scale_factor, self.scale_factor);
        canvas.reset_transform();
        canvas.set_transform(&transform);
```

I load the roboto regular font

```
       canvas.add_font_mem(&resource!("src/assets/Roboto-Regular.ttf"))
```

I set the following paint for the font

```
        let mut paint = Paint::color(self.style.color.into());
        paint.set_font(&[font]);
        paint.set_font_size(72.0);
```

An I render like this

```

        let lines = canvas.break_text_vec(width, text, &paint)?;

        for line_range in lines {
            if let Ok(text_metrics) =
                canvas.fill_text(self.pos.x as f32, y, &text[line_range], &paint)
            {
                y += font_metrics.height();
            }
```

The problem is, that the `front_metrics.height() = 38)` while the rendered glyphs height is somewhere around 56. The result is that vertical line collide.. Any idea what I am doing wrong?

**Comment by geom3trik** [comment]
Sounds like a scale factor issue. 38 * 1.5 = 57. Are you using a display with 150% scale by any chance? If so, multiply the line height by the scale factor.

**Comment by gabm** [comment]
yes, something is off with the scaling factor... but height is scaling differently than font size - that seems strange to me.. also it seems quantized..

all of it is governed by a canvas scaling factor using a transform (see above).


Uploading recording.mp4…

**Comment by gabm** [comment]
here are some more images

![20240219_14h44m02s_grim](https://github.com/femtovg/femtovg/assets/482105/dc7b9d71-a9c6-4ca8-9ed7-c97eb6cf7c1f)
![20240219_14h43m46s_grim](https://github.com/femtovg/femtovg/assets/482105/2748bcfa-00db-4ef8-97d7-f0c938deabf0)

**Comment by gabm** [comment]
I saw that the demo example uses the actual height.. probably thats the way to go.. it works in this case: https://github.com/femtovg/femtovg/blob/a46ac39b6c497018a1a27f0562db2ade571b7c4f/examples/demo.rs#L377
